api/worker/get.go

```diff
@@ -55,14 +56,21 @@ func GetWorker(c *gin.Context) {
 		"worker": w.GetHostname(),
 	}).Infof("reading worker %s", w.GetHostname())
 
-	w, err := database.FromContext(c).GetWorkerForHostname(ctx, w.GetHostname())
-	if err != nil {
-		retErr := fmt.Errorf("unable to get workers: %w", err)
+	var rBs []*library.Build
 
-		util.HandleError(c, http.StatusNotFound, retErr)
+	for _, b := range w.GetRunningBuilds() {
+		build, err := database.FromContext(c).GetBuild(ctx, b.GetID())
```

Comment:
just wondering, did you consider doing this logic using a table `JOIN` in the database layer?

Reply:
I did think about it.. but not very long, as I didn't really see a good option with the multiple build IDs in a single VARCHAR. This might be a future enhancement TBH. Frustrating that Sqlite3 doesn't support integer arrays, as I think that would have made things a little easier...